shell/platform/common/client_wrapper/encodable_value_unittests.cc

```diff
@@ -283,4 +283,57 @@ TEST(EncodableValueTest, DeepCopy) {
   EXPECT_EQ(std::get<std::string>(innermost_map[EncodableValue("a")]), "b");
 }
 
+// Simple class for testing custom encodable values
+class MockCustomValue {
+ public:
+  MockCustomValue(int x, int y) : x_(x), y_(y) {}
+  ~MockCustomValue() = default;
+
+  int x() const { return x_; }
+  int y() const { return y_; }
+
+ private:
+  int x_;
+  int y_;
+};
+
+TEST(EncodableValueTest, TypeIndexesCorrect) {
+  // Null
+  EXPECT_EQ(EncodableValue().index(), 0u);
+  // Bool
+  EXPECT_EQ(EncodableValue(true).index(), 1u);
+  // Int32
+  EXPECT_EQ(EncodableValue(100).index(), 2u);
+  // Int64
+  EXPECT_EQ(EncodableValue(INT64_C(100)).index(), 3u);
+  // Double
+  EXPECT_EQ(EncodableValue(7.0).index(), 4u);
+  // String
+  EXPECT_EQ(EncodableValue("one").index(), 5u);
+  // ByteList
+  EXPECT_EQ(EncodableValue(std::vector<uint8_t>{0, 1}).index(), 6u);
```

Comment:
Nit: you could simplify all of the collections; there's no reason they need to contain any values. Having a bunch of extra setup just obfuscates the purpose of the test IMO.